### PR TITLE
Core: improve error message for missing "game" entry in yaml

### PR DESCRIPTION
**Pull request by beauxq**
## What is this fixing or adding?

If a yaml has no "game" entry, the message to users is confusing.
`'NoneType' object has no attribute 'lower'`

This changes it to
`Missing "game" entry in yaml`

## How was this tested?

ran generate with the bad yaml and saw error message
ran generate with a good yaml to make sure it works

### CODE REVIEW

File: Generate.py [hunk 1]
```diff
@@ -453,6 +453,10 @@ def roll_settings(weights: dict, plando_options: PlandoOptions = PlandoOptions.b
             raise Exception(f"Option {option_key} has to be in a game's section, not on its own.")
 
     ret.game = get_choice("game", weights)
+    if not isinstance(ret.game, str):
+        if ret.game is None:
+            raise Exception('Missing "game" entry in yaml')
+        raise Exception(f"Invalid game: {ret.game} in yaml")
```
[the remark this answers is not in the document]
Comment (beauxq) [reply Generate.py:459]: This is only if it's not a `str` - and a `str` is needed to give a fuzzy match suggestion.

**Review by Exempt-Medic — APPROVED** [review]
Tested various configurations and the errors were improved. I'm not really sure what "is not a string" covers besides floats/ints, but it does cover those.

**Review by ScipioWright — APPROVED** [review]
Tested, it correctly caught a missing game line.

**Comment by Berserker66** [comment]
I agree the message could be better, I disagree with referring to yaml specifically

**Comment by Exempt-Medic** [comment]
> I agree the message could be better, I disagree with referring to yaml specifically

And you would prefer what? "File {file_name}"?

**Comment by beauxq** [comment]
I thought of using the file name, but I don't see the file name being available in this function.

**Comment by Zannick** [comment]
Maybe `"game" not specified`? `Invalid game: {ret.game}` would be fine for the other case.

**Pull request merged**